contracts/utils/ExpiringMetaCallForwarder.sol

```diff
@@ -0,0 +1,77 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/utils/cryptography/ECDSA.sol";
+import "@openzeppelin/contracts/utils/Address.sol";
+import "@openzeppelin/contracts/utils/cryptography/draft-EIP712.sol";
+import "./SelfMulticall.sol";
+
+/// @notice Contract that forwards expiring meta-calls to ERC2771 contracts
+/// that trust it
+/// @dev The term "meta-call" is used as a subset of "meta-tx" where `value`
+/// is zero and `to` is a contract.
+/// Adapted from OpenZeppelin's MinimalForwarder.sol.
+contract ExpiringMetaCallForwarder is EIP712, SelfMulticall {
+    using ECDSA for bytes32;
+    using Address for address;
+
+    struct ExpiringMetaCall {
+        address from;
+        address to;
+        bytes data;
+        uint256 expirationTimestamp;
```

Comment:
It may potentially reduce the calldata size for `execute()`, I'll check. Depends on if structs in arguments are packed like regular structs or ABI-encoded.

Reply:
I moved `data` after `expirationTimestamp` and ran `test:gas`. The change caused `execute()` to cost ~10-20 more gas. I don't have an explanation :upside_down_face: